### PR TITLE
Dcc Validation Failed Screen (EXPOSUREAPP-8062)

### PR DESCRIPTION
**Pull request by d4rken**
This PR wires up the logic to put the right items in the recyclerview.
There will be a follow up PRs for polishing/finishing the list items.

If you want to test it, see here for some test setup:
https://github.com/corona-warn-app/cwa-app-android/pull/3662/commits/98c0ee67d28712e6738f8633129be231059f009e

But I think a code review suffices and further testing should be done when more elements are wired up and we have better test data.

### CODE REVIEW

File: Corona-Warn-App/src/main/res/layout/covid_certificate_validation_failed_fragment.xml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        </com.google.android.material.appbar.CollapsingToolbarLayout>
+
+    </com.google.android.material.appbar.AppBarLayout>
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/list"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:clipToPadding="false"
+        android:paddingTop="24dp"
```
Comment (jurajkusnier) [inline Corona-Warn-App/src/main/res/layout/covid_certificate_validation_failed_fragment.xml:53]: I removed horizontal margin for individual items. Let's setup it here. 
```suggestion
        android:paddingHorizontal="@dimen/spacing_small"
```

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/ui/validationresult/failed/DccValidationFailedFragment.kt [hunk 2]
```diff
@@ -1,5 +1,43 @@
...
+            factory.create(
+                validation = args.validation
+            )
+        }
+    )
+
+    @Inject lateinit var validationResultAdapter: ValidationResultAdapter
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        binding.list.adapter = validationResultAdapter
```
Comment (jurajkusnier) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/ui/validationresult/failed/DccValidationFailedFragment.kt:36]: I removed vertical padding for individual elements, we can rather use `ItemDecoration()`
```suggestion
        binding.list.apply {
            adapter = validationResultAdapter
            addItemDecoration(
                RecylerViewPaddingDecorator(
                    topPadding = R.dimen.spacing_small
                )
            )
        }
```
`RecylerViewPaddingDecorator` is available in #3646

File: Corona-Warn-App/src/main/res/drawable-night/covid_certificate_validation_failed_header.xml [hunk 3]
```diff
@@ -0,0 +1,116 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (jurajkusnier) [inline Corona-Warn-App/src/main/res/drawable-night/covid_certificate_validation_failed_header.xml:1]: The header has been changed in Figma

File: Corona-Warn-App/src/main/res/layout/covid_certificate_validation_failed_fragment.xml (hunk 1 above)
Comment (d4rken) [reply Corona-Warn-App/src/main/res/layout/covid_certificate_validation_failed_fragment.xml]: I added it via the decorator.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/ui/validationresult/common/CovidCertificateValidationResultFragmentsBindingExtensions.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        // TODO
+    }
+    DccValidation.State.OPEN -> {
+        // TODO
+    }
+    DccValidation.State.TECHNICAL_FAILURE,
+    DccValidation.State.FAILURE -> {
+        toolbar.setTitle(R.string.validation_failed_title)
+        headerImage.setImageResource(R.drawable.covid_certificate_validation_failed_header)
+    }
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/ui/validationresult/common/CovidCertificateValidationResultFragmentsBindingExtensions.kt:18]: Needs to be filled out in other PRs.

/cc @BMItr

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/ui/validationresult/failed/DccValidationFailedFragment.kt [hunk 5]
```diff
@@ -1,5 +1,58 @@
...
+                validation = args.validation
+            )
+        }
+    )
+
+    @Inject lateinit var validationResultAdapter: ValidationResultAdapter
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        binding.apply {
+            setHeaderForState(args.validation.state)
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/ui/validationresult/failed/DccValidationFailedFragment.kt:39]: @BMItr Use the same layout file and then just set the state :beers:.

**Review by BMItr — DISMISSED** [review]
perfect!!! ❤️

**Review by chris-cwa — DISMISSED** [review]

**Review by jurajkusnier — DISMISSED** [review]

File: Corona-Warn-App/src/main/res/values-de/covid_certificate_strings.xml [hunk 6]
```diff
@@ -213,4 +213,7 @@
     <!-- XTXT: Validation Rule Technical Failure - Date Expired  -->
     <string name="validation_rule_technical_error_date_expired">Das Zertifikat darf nicht abgelaufen sein.</string>
 
+    <!-- Validation Failed Screen-->
+    <!-- XHED: Toolbar title for the validation failed screen -->
+    <string name="validation_failed_title">Kein gültiges Zertifikat</string>
```
Comment (NataliaLemmerth) [inline Corona-Warn-App/src/main/res/values-de/covid_certificate_strings.xml:218]:
```suggestion
    <string name="validation_failed_title">Zertifikat nicht gültig</string>
```
according to the current state in Figma

File: Corona-Warn-App/src/main/res/values/covid_certificate_strings.xml [hunk 7]
```diff
@@ -215,4 +215,7 @@
     <!-- XTXT: Validation Rule Technical Failure - Date Expired  -->
     <string name="validation_rule_technical_error_date_expired">Das Zertifikat darf nicht abgelaufen sein.</string>
 
+    <!-- Validation Failed Screen-->
+    <!-- XHED: Toolbar title for the validation failed screen -->
+    <string name="validation_failed_title">Kein gültiges Zertifikat</string>
```
Comment (NataliaLemmerth) [inline Corona-Warn-App/src/main/res/values/covid_certificate_strings.xml:220]:
```suggestion
    <string name="validation_failed_title">Zertifikat nicht gültig</string>
```

**Review by NataliaLemmerth — CHANGES_REQUESTED** [review]
Please change the text to "Zertifikat nicht gültig".

**Comment by BMItr** [comment]
@d4rken close button listener for chrismas?

**Review by BMItr — APPROVED** [review]

**Review by NataliaLemmerth — APPROVED** [review]
UA approved

**Review by chris-cwa — APPROVED** [review]
LGTM!

**Pull request merged**